src/Components/Game/WordDisplay/Timer.tsx

```diff
@@ -19,25 +43,25 @@ const Timer = ({ startTime }: OwnPropsT) => {
     if(timerRef.current) clearInterval(timerRef.current)
     
     timerRef.current = setInterval(() => {
-      if(timeRef.current === 0) clearInterval(timerRef.current)
-      else setTime((prevTime) => prevTime - 1)
+      if(timeRef.current === 0) {
+        if(drawer && sessionId === drawer.sessionId) socket.emit('next_drawer', roomName)
+        clearInterval(timerRef.current)
+      }
+      else setTime((prevTime) => {
+        console.log(prevTime - 1)
+        return prevTime - 1
+      })
     }, 1000)
   }
 
-  //@ts-ignore
-  const updateTimer = (newTime: number) => {
-      setTime(newTime)
-      startTimer()
+  const updateTimer = (updatedTime: number) => {
+    setTime(updatedTime)
+    startTimer()
   }
 
   const resetTimer = () => {
-    if(timerRef.current) clearInterval(timerRef.current)
-    setTime(startTime)
-    
-    timerRef.current = setInterval(() => {
-      if(timeRef.current === 0) clearInterval(timerRef.current)
-      else setTime((prevTime) => prevTime - 1)
-    }, 1000)
+    setTime(START_TIME)
+    startTimer()
```

Comment:
The reset timer doesn't reset the backend timer, so even though you can set it back to original start, at the 5 sec intervals, the backend timer will bring the time back in sync with it.